### DISCUSSION TITLE
Integrate LM Studio models

### DISCUSSION THREAD

**Discussion by iamlasse** [cline/cline#376] [Feature Requests]
I have been testing this extension for a while, and the ollama models don’t work or are buggy, would it be possible to integrate lm studio models as there are more available it seems?

**Comment by celltower** [comment]
same for me ..

**Comment by dcbartlett** [comment]
LM Studio is supported as of 2.1.6.  Because LM Studio supports many different models, We can only provide access for you to configure LM Studio in the models list.

Here is an example of a working configuration using LM Studio
![image](https://github.com/user-attachments/assets/d9c9e5b2-44b5-4248-a49b-303432363ed1)

**Reply by moniljo** [reply]
Deepseek R1 doesn't seem to work with cline, even though it works fine with Lm Studio chat mode. Seems to be stuck waiting for a response.

2025-01-21 13:40:31  [INFO] 
[LM STUDIO SERVER] Running chat completion on conversation with 2 messages.
2025-01-21 13:40:31  [INFO] 
[LM STUDIO SERVER] Streaming response...
2025-01-21 13:40:31 [DEBUG] 
Sampling params:	repeat_last_n = 64, repeat_penalty = 1.100, frequency_penalty = 0.000, presence_penalty = 0.000
	dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
	top_k = 40, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, temp = 0.000
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
2025-01-21 13:40:31 [DEBUG] 
sampling: 
logits -> logit-bias -> penalties -> dry -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist 
generate: n_ctx = 128000, n_batch = 512, n_predict = -1, n_keep = 0
2025-01-21 13:40:45 [DEBUG]  PromptProcessing: 57.0848
2025-01-21 13:40:59 [DEBUG] PromptProcessing: 58.7637
2025-01-21 13:41:13 [DEBUG] PromptProcessing: 60.4427
2025-01-21 13:41:27 [DEBUG] PromptProcessing: 62.1217
2025-01-21 13:41:42 [DEBUG] PromptProcessing: 63.8006
2025-01-21 13:41:57 [DEBUG] PromptProcessing: 65.4796
2025-01-21 13:42:12 [DEBUG] PromptProcessing: 67.1585
2025-01-21 13:42:28 [DEBUG] PromptProcessing: 68.8375
2025-01-21 13:42:44 [DEBUG] PromptProcessing: 70.5165
2025-01-21 13:42:59 [DEBUG] PromptProcessing: 72.1954

**Comment by Amari-Mecheri** [comment]
I tried deepseek r1 14b. I asked 1+1. I had to set tokens to 16384 to have it working. I don't know if the fact my project was opened had an impact. To answer the question R1 did some reasoning about how it should answer the question, concluded it was not allowed to answer directly but had to use the tools so it requested to use a python command. I accepted and the command was ran in the terminal and displayed 2. However deepseek resumed reasoning and once again concluded it had to use a command which was allowed and gave the same result in the terminal. The model again got into a reasoning phase which led to the same command. I finally canceled the task.
